spec/controllers/settings/profiles_controller_spec.rb

```diff
@@ -23,5 +23,10 @@
       expect(account.reload.display_name).to eq 'New name'
       expect(response).to redirect_to(settings_profile_path)
     end
+
+    it 'renders :show if parameter is missing' do
+      put :update
+      expect(response).to render_template :show
```

Comment:
I think it's fine if they are different, because they are different things.

When the params are missing, it's very similar to a missing route or some other request which can't be fulfilled, so I'd expect a 4xx response (and rails catches the params error and does this, I think?) An exception to this would be if there are multiple ways the route could be valid and different params combination were optional or valid depending on the presence of other ones ... I'd expect something more graceful there.

When the params/route is valid - but the data included in the update is invalid - I'd expect a 200 response with a rendered page that includes a description of the data validation errors.